flask/src/main/java/net/corda/gradle/flask/FlaskJarTask.java

```diff
@@ -0,0 +1,235 @@
+            String entryName = fileCopyDetailsInternal.getRelativePath().toString();
+            if(!fileCopyDetailsInternal.isDirectory() && entryName.startsWith(Flask.Constants.LIBRARIES_FOLDER)) {
+                md.reset();
+                try(InputStream is = new DigestInputStream(new FileInputStream(fileCopyDetailsInternal.getFile()), md)) {
+                    while(is.read(buffer) >= 0) {}
+                }
+                Attributes attr = manifest.getEntries().computeIfAbsent(entryName, it -> new Attributes());
+                attr.putValue(Flask.ManifestAttributes.ENTRY_HASH,
+                        String.format("%032x", new BigInteger(1, md.digest())));
+            }
+            fileCopyDetailsInternals.add(fileCopyDetailsInternal);
+        }
+    }
+
+    @Override
+    protected CopyAction createCopyAction() {
+        File destination = getArchiveFile().get().getAsFile();
+        return new CopyAction() {
+            @Override
+            @SneakyThrows
+            public WorkResult execute(CopyActionProcessingStream copyActionProcessingStream) {
+                try (ZipOutputStream zipOutputStream = new ZipOutputStream(new FileOutputStream(destination))) {
+                    Manifest manifest = new Manifest();
+                    manifest.getMainAttributes().put(Attributes.Name.MANIFEST_VERSION, "1.0");
+                    manifest.getMainAttributes().put(Attributes.Name.MAIN_CLASS, launcherClassName.get());
+                    Optional.ofNullable(mainClassName.getOrNull()).ifPresent(it ->
+                        manifest.getMainAttributes().putValue(Flask.ManifestAttributes.APPLICATION_CLASS, it));
+                    Optional.ofNullable(jvmArgs.getOrNull())
+                            .filter(it -> !it.isEmpty())
+                            .ifPresent(it -> manifest.getMainAttributes().putValue(Flask.ManifestAttributes.JVM_ARGS,
+                                    ManifestEscape.escapeStringList(it)));
+                    MessageDigest md5 = MessageDigest.getInstance("MD5");
+                    byte[] buffer = new byte[0x10000];
+                    if(!javaAgents.isEmpty()) {
+                        List<String> agentsStrings = javaAgents.stream().map(new Function<JavaAgent, String>() {
+                            @Override
+                            @SneakyThrows
+                            public String apply(JavaAgent javaAgent) {
+                                md5.reset();
+                                try (InputStream stream = new DigestInputStream(new FileInputStream(javaAgent.jar), md5)) {
+                                    while (stream.read(buffer) >= 0) {
+                                    }
+                                }
+                                StringBuilder sb = new StringBuilder();
+                                sb.append(String.format("%032x", new BigInteger(1, md5.digest())));
```

Comment:
_Definitely_ a candidate for a utility function!

Reply:
ack